micrometer-core/src/main/java/io/micrometer/core/instrument/binder/jvm/JvmThreadMetrics.java

```diff
@@ -66,6 +66,11 @@ public void bindTo(MeterRegistry registry) {
                 .description("The current number of live threads including both daemon and non-daemon threads")
                 .baseUnit(BaseUnits.THREADS).register(registry);
 
+        Gauge.builder("jvm.threads.started", threadBean, ThreadMXBean::getTotalStartedThreadCount).tags(tags)
```

Comment:
This should be monotonic and I guess what users are mostly interested in is not the total number of threads that has been started since the JVM started but the rate of them. So I think this must be a counter. Since there is no event you can use to increment a "normal" `Counter` but there is a (monothonic) method that gives you the current total, you can use a `FunctionCounter` instead of a `Gauge`, see the [docs](https://micrometer.io/docs/concepts#_function_tracking_counters).

Reply:
Makes sense. But still worried about Step Registry users like us. But it really makes sense to have this as a Counter.

Reply:
Check step registries, for those this will return delta not cumulative.